lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -58,6 +58,12 @@ pub struct EnforcingSigner {
 	pub disable_revocation_policy_check: bool,
 }
 
+impl PartialEq for EnforcingSigner {
+	fn eq(&self, o: &Self) -> bool {
+		Arc::ptr_eq(&self.state, &o.state)
```

Comment:
It's interesting when I checked out this function and went down a bit of a rabbit hole and found out that currently for `dyn Trait` raw pointers, Rust also checks for equality of vtables which has caused what most would naturally consider false positives as it was unexpected from the docs: https://github.com/rust-lang/rust/issues/106447

We're at least good for `EnforcementState` structs though.